golem-component-service-base/src/repo/component.rs

```diff
@@ -0,0 +1,510 @@
+            .bind(component.protected_component.clone())
+            .bind(component.protector_version)
+            .bind(component.metadata.clone())
+            .execute(&mut *transaction)
+            .await?;
+
+        transaction.commit().await?;
+
+        Ok(())
+    }
+
+    async fn get(
+        &self,
+        namespace: &str,
+        component_id: &Uuid,
+    ) -> Result<Vec<ComponentRecord>, RepoError> {
+        sqlx::query_as::<_, ComponentRecord>(
+            "SELECT components.namespace, components.name, component_versions.* FROM components JOIN component_versions ON components.component_id = component_versions.component_id WHERE components.namespace = $1 AND component_versions.component_id = $2"
+        )
+            .bind(namespace)
+            .bind(component_id)
+            .fetch_all(self.db_pool.deref())
+            .await
+            .map_err(|e| e.into())
+    }
+
+    async fn get_all(&self, namespace: &str) -> Result<Vec<ComponentRecord>, RepoError> {
+        sqlx::query_as::<_, ComponentRecord>("SELECT components.namespace, components.name, component_versions.* FROM components JOIN component_versions ON components.component_id = component_versions.component_id WHERE components.namespace = $1")
+            .bind(namespace)
+            .fetch_all(self.db_pool.deref())
+            .await
+            .map_err(|e| e.into())
+    }
+
+    async fn get_latest_version(
+        &self,
+        namespace: &str,
+        component_id: &Uuid,
+    ) -> Result<Option<ComponentRecord>, RepoError> {
+        sqlx::query_as::<_, ComponentRecord>(
+            "SELECT components.namespace, components.name, component_versions.* FROM components JOIN component_versions ON components.component_id = component_versions.component_id WHERE components.namespace = $1 AND component_versions.component_id = $2  ORDER BY version DESC LIMIT 1",
+        ).bind(namespace)
+            .bind(component_id)
+            .fetch_optional(self.db_pool.deref())
+            .await
+            .map_err(|e| e.into())
+    }
+
+    async fn get_by_version(
+        &self,
+        namespace: &str,
+        component_id: &Uuid,
+        version: u64,
+    ) -> Result<Option<ComponentRecord>, RepoError> {
+        sqlx::query_as::<_, ComponentRecord>(
+            "SELECT components.namespace, components.name, component_versions.* FROM components JOIN component_versions ON components.component_id = component_versions.component_id WHERE components.namespace = $1 AND component_versions.component_id = $2 AND component_versions.version = $3",
+        )
+            .bind(namespace)
+            .bind(component_id)
+            .bind(version as i64)
+            .fetch_optional(self.db_pool.deref())
+            .await
+            .map_err(|e| e.into())
+    }
+
+    async fn get_by_name(
+        &self,
+        namespace: &str,
+        name: &str,
+    ) -> Result<Vec<ComponentRecord>, RepoError> {
```

Comment:
It's not a `Vec` - it should be `Option`

Reply:
fixed